src/main/java/com/sprint/mission/discodeit/config/SecurityConfig.java

```diff
@@ -0,0 +1,179 @@
+import org.springframework.security.web.authentication.rememberme.PersistentTokenRepository;
+import org.springframework.security.web.context.HttpSessionSecurityContextRepository;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.context.SecurityContextRepository;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+
+@Configuration
+@EnableWebSecurity //  Spring Security를 활성화하며, 내부적으로 WebSecurityConfigurerAdapter 없이도 커스터마이징이 가능하게 설정합니다.
+@AllArgsConstructor
+@EnableMethodSecurity(prePostEnabled = true)
+public class SecurityConfig {
+  private final UserDetailsService userDetailsService;
+  private final DataSource dataSource;
+  private final JwtSessionService jwtSessionService;
+  @Bean
+  public SecurityFilterChain securityFilterChain(
+      HttpSecurity http,
+      AuthenticationManager authManager,
+      SecurityContextRepository contextRepo,
+      SessionRegistry sessionRegistry) throws Exception {
+
+    CustomUsernamePasswordAuthenticationFilter loginFilter =
+        new CustomUsernamePasswordAuthenticationFilter(authManager, sessionRegistry, objectMapper());
+    loginFilter.setSecurityContextRepository(contextRepo);
+
+    //AntPathRequestMatcher 키워드
+    List<RequestMatcher> publicMatchers = List.of(
+        new AntPathRequestMatcher("/api/auth/csrf-token", HttpMethod.GET.name()),
+        new AntPathRequestMatcher("/api/auth/login", HttpMethod.POST.name()),
+        new AntPathRequestMatcher("/api/users", HttpMethod.POST.name())
+    );
+
+    List<RequestMatcher> adminMatchers = List.of(
+        new AntPathRequestMatcher("/api/users/role", HttpMethod.PUT.name())
+    );
+
+    List<RequestMatcher> channelManagerMatchers = List.of(
+        new AntPathRequestMatcher("/api/channels/public/**", HttpMethod.POST.name()),
+        new AntPathRequestMatcher("/api/channels/public/**", HttpMethod.PUT.name()),
+        new AntPathRequestMatcher("/api/channels/public/**", HttpMethod.DELETE.name())
+    );
+
+    RequestMatcher userMatcher = new AntPathRequestMatcher("/api/**");
+
+    http
+        .csrf(csrf -> csrf.csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse()))
+        .authorizeHttpRequests(auth -> {
+          publicMatchers.forEach(matcher -> auth.requestMatchers(matcher).permitAll());
+          adminMatchers.forEach(matcher -> auth.requestMatchers(matcher).hasRole("ADMIN"));
+          channelManagerMatchers.forEach(matcher -> auth.requestMatchers(matcher).hasRole("CHANNEL_MANAGER"));
+          auth.requestMatchers(userMatcher).hasRole("USER");
+          auth.anyRequest().permitAll();
+        })
```

Comment:
JWT 토큰 기반 인증으로 리팩토링한다면 기존의 세션은 인증 과정에서 불필요하므로 아래 설정을 넣어주는게 좋을 것 같아요.
```suggestion
        })
        .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
```